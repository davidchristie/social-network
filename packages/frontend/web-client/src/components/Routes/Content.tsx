import React from "react";
import { Redirect, Route, Switch } from "react-router-dom";
import Page from "../Page";
import {
  defaultAuthenticatedPath,
  defaultUnauthenticatedPath,
  routes
} from "./constants";

export interface Props {
  authenticated: boolean;
}

const Content: React.StatelessComponent<Props> = ({ authenticated }) => {
  const defaultPath = authenticated
    ? defaultAuthenticatedPath
    : defaultUnauthenticatedPath;
  const filteredRoutes = routes
    .filter(route => {
      return authenticated
        ? !route.onlyUnauthenticated
        : !route.onlyAuthenticated;
    });
  return (
    <Switch>
      {filteredRoutes.map(route => {
        return (
          <Route
            exact={true}
            key={route.path}
            path={route.path}
            render={() => <Page load={route.load} />}
          />
        );
      })}
      <Route
        path="*"
        render={() => <Redirect to={defaultPath} />}
      />
    </Switch>
  );
};

export default Content;
