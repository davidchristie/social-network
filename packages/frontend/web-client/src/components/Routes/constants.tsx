import React from "react";

export interface RouteData {
  load: () => Promise<{
    default: React.ComponentType
  }>;
  onlyAuthenticated?: boolean;
  onlyUnauthenticated?: boolean;
  path: string;
}

export const defaultAuthenticatedPath = "/";
export const defaultUnauthenticatedPath = "/login";

export const routes: RouteData[] = [
  {
    load: () => import(/* webpackChunkName: "HomePage" */ "../HomePage"),
    onlyAuthenticated: true,
    path: "/",
  },
  {
    load: () => import(/* webpackChunkName: "AccountPage" */ "../AccountPage"),
    onlyAuthenticated: true,
    path: "/account",
  },
  {
    load: () => import(/* webpackChunkName: "LoginPage" */ "../LoginPage"),
    onlyUnauthenticated: true,
    path: "/login",
  },
  {
    load: () => import(/* webpackChunkName: "ProfilePage" */ "../ProfilePage"),
    path: "/profile/:id",
  },
  {
    load: () => import(/* webpackChunkName: "SignupPage" */ "../SignupPage"),
    onlyUnauthenticated: true,
    path: "/signup",
  },
];
