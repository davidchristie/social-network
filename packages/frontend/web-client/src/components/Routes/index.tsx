import { Authenticated, Unauthenticated } from "domain-model";
import React from "react";
import { Redirect, Route, Switch } from "react-router-dom";
import {
  defaultAuthenticatedPath,
  defaultUnauthenticatedPath,
  routes
} from "./constants";

const Routes: React.StatelessComponent = () => {
  const authenticatedRoutes = routes.filter(route => !route.onlyUnauthenticated);
  const unauthenticatedRoutes = routes.filter(route => !route.onlyAuthenticated);
  return (
    <React.Fragment>
      <Authenticated>
        <Switch>
          {authenticatedRoutes.map(route => {
            return (
              <Route
                component={route.component}
                exact={true}
                key={route.path}
                path={route.path}
              />
            );
          })}
          <Route
            path="*"
            render={() => <Redirect to={defaultAuthenticatedPath} />}
          />
        </Switch>
      </Authenticated>
      <Unauthenticated>
        <Switch>
          {unauthenticatedRoutes.map(route => {
            return (
              <Route
                component={route.component}
                exact={true}
                key={route.path}
                path={route.path}
              />
            );
          })}
          <Route
            path="*"
            render={() => <Redirect to={defaultUnauthenticatedPath} />}
          />
        </Switch>
      </Unauthenticated>
    </React.Fragment>
  );
};

export default Routes;
