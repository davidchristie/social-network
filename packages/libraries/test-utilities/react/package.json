{
  "main": "../dist/react/index.js",
  "name": "test-utilities/react",
  "private": true,
  "version": "1.0.0"
}
