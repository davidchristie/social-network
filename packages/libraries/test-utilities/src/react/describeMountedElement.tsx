import { mount } from "enzyme";
import React from "react";

export default function describeMountedElement (
  name: string,
  element: React.ReactElement<any>
) {
  describe(name, () => {
    it("renders without errors", () => {
      mount(element);
    });

    it("matches shapshot", () => {
      const wrapper = mount(element);
      expect(wrapper).toMatchSnapshot();
    });
  });
}
