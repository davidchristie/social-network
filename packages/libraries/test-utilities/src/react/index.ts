export { default as describeMountedElement } from "./describeMountedElement";
export { default as describeWithCustomClass } from "./describeWithCustomClass";
export { default as describeWithNoProps } from "./describeWithNoProps";
export { default as describeWithProps } from "./describeWithProps";
export { default as itMatchesSnapshot } from "./itMatchesSnapshot";
