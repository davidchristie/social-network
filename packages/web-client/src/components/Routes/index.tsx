import { Alert } from "design-system";
import React from "react";
import { Query } from "react-apollo";
import { Redirect, Route, Switch } from "react-router-dom";

import AccountQuery, {
  AccountData,
  AccountVariables,
} from "../../queries/Account";
import AccountPage from "../AccountPage";
import HomePage from "../HomePage";
import LoginPage from "../LoginPage";
import ProfilePage from "../ProfilePage";
import SignupPage from "../SignupPage";

export default class Routes extends React.Component {
  public render () {
    return (
      <Query<AccountData, AccountVariables>
        query={AccountQuery}
      >
        {({ data, error, loading }) => {
          if (loading) {
            return "Loading";
          }
          if (error) {
            return <Alert>{error.message}</Alert>;
          }
          return data!.account
            ? (
              <Switch>
                <Route
                  component={HomePage}
                  exact={true}
                  path="/"
                />
                <Route
                  component={AccountPage}
                  exact={true}
                  path="/account"
                />
                <Route
                  component={ProfilePage}
                  exact={true}
                  path="/profile/:id"
                />
                <Route
                  path="*"
                  render={() => <Redirect to="/" />}
                />
              </Switch>
            )
            : (
              <Switch>
                <Route
                  component={LoginPage}
                  exact={true}
                  path="/login"
                />
                <Route
                  component={SignupPage}
                  exact={true}
                  path="/signup"
                />
                <Route
                  path="*"
                  render={() => <Redirect to="/login" />}
                />
              </Switch>
            );
        }}
      </Query>
    );
  }
}
