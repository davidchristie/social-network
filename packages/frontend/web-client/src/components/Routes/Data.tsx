import { Alert, Loading } from "design-system";
import React from "react";
import AccountQuery from "../AccountQuery";
import { Props as ContentProps } from "./Content";

export interface Props {
  content: React.ComponentType<ContentProps>;
}

const Data: React.StatelessComponent<Props> = ({ content }) => {
  const Content = content;
  return (
    <AccountQuery>
      {({ data, error, loading }) => {
        if (error) {
          return <Alert>{error.message}</Alert>;
        }
        if (!data || loading) {
          return <Loading />;
        }
        const authenticated = Boolean(data.account);
        return <Content authenticated={authenticated} />;
      }}
    </AccountQuery>
  );
};

export default Data;
