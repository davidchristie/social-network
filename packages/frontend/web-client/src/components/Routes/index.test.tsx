import React from "react";
import { itMatchesSnapshot } from "test-utilities/dist/react";
import Routes from ".";

describe("Routes component", () => {
  itMatchesSnapshot(<Routes />);
});
