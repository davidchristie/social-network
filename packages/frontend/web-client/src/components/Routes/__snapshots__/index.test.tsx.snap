// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes component matches snapshot 1`] = `
<Fragment>
  <Authenticated>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        key="/"
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        key="/account"
        path="/account"
      />
      <Route
        component={[Function]}
        exact={true}
        key="/profile/:id"
        path="/profile/:id"
      />
      <Route
        path="*"
        render={[Function]}
      />
    </Switch>
  </Authenticated>
  <Unauthenticated>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        key="/login"
        path="/login"
      />
      <Route
        component={[Function]}
        exact={true}
        key="/profile/:id"
        path="/profile/:id"
      />
      <Route
        component={[Function]}
        exact={true}
        key="/signup"
        path="/signup"
      />
      <Route
        path="*"
        render={[Function]}
      />
    </Switch>
  </Unauthenticated>
</Fragment>
`;
